.github/actions/BuildCirct/action.yml

```diff
@@ -34,3 +35,11 @@ runs:
           --install-path ${{ github.workspace }}/build-circt/circt \
           --llvm-lit-path ~/.local/bin/lit
       shell: bash
+
+    - name: "Save CIRCT to the cache"
+      id: save-cache-circt
+      uses: actions/cache/save@v4
```

Comment:
This should only happen if we did not have a cache hit in the first place, right? In other words, we do not need to save anything if we found it in the cache.

Reply:
The example I followed didn't make the check. It's more or less instantaneous so doesn't really hurt.

Reply:
I was not concerned about performance, but more about the fact that you are always overwriting the old cache even if it there was a hit in the cache.

Reply:
I simply follow the example given by the cache action

https://github.com/actions/cache/blob/main/README.md#example-cache-workflow

As far as what I understand, if the save finds the key in the cache, then the cache is not updated.

Reply:
My understanding is that the rebuilding of CIRCT should be skipped when there is a cache hit: https://github.com/actions/cache/blob/main/README.md#skipping-steps-based-on-cache-hit

However, I only skimmed the documentation, so I might have misunderstood.